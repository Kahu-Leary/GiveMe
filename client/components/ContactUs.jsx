import React from 'react'
import emailjs from 'emailjs-com'

export default function ContactUs () {
  function sendEmail (e) {
    e.preventDefault()

    emailjs.sendForm('service_r9l20s7', 'template_7tw16ho', e.target, 'user_Rj2yKyFEMTlfma4c44jwR')
      .then((result) => {
        return console.log(result.text)
      }).catch(error => {
        return console.log(error.text)
      })
    e.target.reset()
  }

  return (
    <>
      <div className="form">
        <div className="formLeft">
          <img src="./img/logo-txt-orange.png" alt="logo" className="logoImg"/>
        </div>
        <div className="formRight">
          <h1>Contact Us</h1>
          <form onSubmit={sendEmail}>

            <input type="text" className="formInput" placeholder="Name" name="name" />

            <input type="email" className="formInput" placeholder="Email Address" name="email" />

            <input type="text" className="formInput" placeholder="Subject" name="subject" />

            <textarea className="textarea" id="" cols="30" rows="8" placeholder="Your message" name="message"></textarea>

            <input type="submit" className="button primaryBtn" value="Send Message"></input>
          </form>
        </div>

      </div>
    </>
  )
}
