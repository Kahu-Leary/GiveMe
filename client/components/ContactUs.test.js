import React from 'react'

import renderer from 'react-test-renderer'

import ContactUs from './ContactUs'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ContactUs/>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
