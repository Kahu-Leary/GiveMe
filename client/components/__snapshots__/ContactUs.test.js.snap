// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="form"
>
  <div
    className="formLeft"
  >
    <img
      alt="logo"
      className="logoImg"
      src="./img/logo-txt-orange.png"
    />
  </div>
  <div
    className="formRight"
  >
    <h1>
      Contact Us
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        className="formInput"
        name="name"
        placeholder="Name"
        type="text"
      />
      <input
        className="formInput"
        name="email"
        placeholder="Email Address"
        type="email"
      />
      <input
        className="formInput"
        name="subject"
        placeholder="Subject"
        type="text"
      />
      <textarea
        className="textarea"
        cols="30"
        id=""
        name="message"
        placeholder="Your message"
        rows="8"
      />
      <input
        className="button primaryBtn"
        type="submit"
        value="Send Message"
      />
    </form>
  </div>
</div>
`;
